patina_dxe_core/src/hw_interrupt_protocol.rs

```diff
@@ -451,20 +451,23 @@ impl HwInterruptProtocolHandler {
     }
 }
 
-#[derive(IntoComponent, Default)]
+#[derive(IntoComponent)]
 /// A component to install the two hardware interrupt protocols.
-pub(crate) struct HwInterruptProtocolInstaller;
+pub(crate) struct HwInterruptProtocolInstaller(GicBases);
 
 impl HwInterruptProtocolInstaller {
+    pub fn new(gic_bases: GicBases) -> Self {
+        Self(gic_bases)
+    }
+
     fn entry_point(
         self,
         interrupt_manager: Service<dyn InterruptManager>,
-        gic_bases: Config<GicBases>,
         boot_services: StandardBootServices,
     ) -> patina::error::Result<()> {
-        log::info!("GICv3 initializing {:x?}", (gic_bases.0, gic_bases.1));
+        log::info!("GICv3 initializing {:x?}", (self.0.0, self.0.1));
```

Comment:
This is starting to look like an ip address. 

Can we maybe change this to 
`pub struct GicBases {
  gic_distributor: u64,
  gic_redistributor: u64
}`
so we have some nice self-documenting names? Could be done as a separate PR if too off-topic.

Reply:
lmao I agree this is a bit much. I'll make an issue and do it as a follow up PR